### ISSUE TITLE
Number of iterations different than stated

### ISSUE THREAD

**Issue by CIWa** [marbl/SALSA#151]
Hi,
I would like to report that Salsa still seems to sometimes run more iterations than stated in the call. More precisely, I requested the default number of iterations (3), but Salsa ran 5. 
My call was:

`run_pipeline.py -a assembly.fa -l assembly.fa.fai -b alignment.bed -o ./ -e DNASE -m yes`

Could be the same problem as in issue #29?
Best wishes,
Isabel

**Comment by oneillkza** [comment]
Yep I'm having the same issue -- SALSA always runs 5 iterations (well, maybe 4 -- it looks like "iteration 1" is just the baseline assembly). This is different from the default of 3 mentioned in the docs, and seems to ignore the `-i` parameter completely.

It even seems to be slightly stochastic -- one time I ran it with `-i 7` it ran for five iterations. Another time it ran for four.

**Comment by skoren** [comment]
@ghuryejay can confirm but I'm pretty sure the iteration parameter is no longer used. As described in the manuscript for SALSA2, the stopping condition is dynamically determined based on when the majority of new links are bad or the assembly is not changing.

**Comment by oneillkza** [comment]
Thanks for the explanation @skoren ! It might be good to update the documentation/remove the parameter.

And is there any way to fine-tune when it sops? I have an assembly I've scaffolded with SALSA, which from the final contact map looks very good, but it's visually clear that there are a few more scaffolds that could be joined.
